brvs-core/src/main/java/iroha/validation/rest/RestService.java

```diff
@@ -495,20 +493,19 @@ private void checkTransactionSignaturesCount(Transaction transaction) {
       ToriiResponse res = handler.apply(requestedTx);
       String status = printer.print(res);
       logger.info("Got transaction status {}", status);
-      return Response.status(HttpStatus.SC_OK).entity(status).build();
+      return Response.ok(new ToriiResultResponse(res)).build();
     } catch (InvalidProtocolBufferException | IllegalArgumentException e) {
       logger.error("Error during transaction processing", e);
-      return Response.status(HttpStatus.SC_UNPROCESSABLE_ENTITY).build();
+      throw new BrvsException(e.getMessage(), e, MALFORMED_TRANSACTION);
     } catch (Exception e) {
-      logger.error("Error during transaction processing", e);
-      return Response.status(HttpStatus.SC_INTERNAL_SERVER_ERROR).build();
+      throw new RuntimeException(e);
```

Comment:
I cannot omit the block since `handler.apply` throws `Exception` so I just rethrow it. Btw I don't know why there is a general `Exception` at the interface declaration, will make it more specific.